onnxruntime/python/backend/backend.py

```diff
@@ -107,7 +107,19 @@ def prepare(cls, model, device=None, **kwargs):
             for k, v in kwargs.items():
                 if hasattr(options, k):
                     setattr(options, k, v)
-            inf = InferenceSession(model, sess_options=options, providers=get_available_providers())
+
+            providers = get_available_providers()
+
+            if os.getenv('EXCLUDE_PROVIDERS'):
+                exclude_providers = os.getenv('EXCLUDE_PROVIDERS').split(',')
+            else:
+                exclude_providers = []
+
+            # exclude TRT EP temporarily and only test CUDA EP to retain previous behavior
+            if 'TensorrtExecutionProvider' in providers and 'TensorrtExecutionProvider' in exclude_providers:
```

Comment:
remove this entire block for checking TRT EP. only exclude via EXCLUDE_PROVIDERS

Reply:
the logic should be to remove excluded providers from the list returned by get_available_providers()
then you don't need to hardcode or specifically mention any providers by name.